app/Models/Airline.php

```diff
@@ -11,11 +11,39 @@ class Airline extends Model
 {
     protected $guarded = ['id'];
 
-    public function flights(): HasMany {
+    public function flights(): HasMany
+    {
         return $this->hasMany(Flight::class);
     }
 
-    public function cities(): BelongsToMany {
+    public function cities(): BelongsToMany
+    {
         return $this->belongsToMany(City::class);
     }
+
+    public function scopeOrder(Builder $query, ?string $sortBy, bool $ascending): void
+    {
+        $allowedColumns = ['id', 'name'];
+        $order = $ascending ? 'asc' : 'desc';
+
+        $sortBy = in_array($sortBy, $allowedColumns)? $sortBy: 'id';
+
+        $query->orderBy($sortBy, $order);
+    }
+
+    public function scopeFilterByCity(Builder $query, ?string $city): void
+    {
+        $query->when($city, function($query, $city) {
+            $query->whereHas('cities', function($query) use($city) {
+                $query->where('city_id', $city);
+            });
+        });
+    }
+
+    public function scopeFilterByFlights(Builder $query, ?int $flights): void
+    {
+        $query->when(isset($flights), function ($query) use ($flights) {
+            $query->having('flights_count', '=', $flights);
+        });
+    }
```

Comment:
If they are ids, add Id to the variable name, that way you remember it's an Id and not the Model